source/develop/release-management/features/virt/migration-encryption.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Migration encryption
+category: feature
+authors: mzamazal, ljelinko
+feature_name: Migration encryption
+feature_modules: engine, vdsm
+---
+
+# Migration encryption
+
+## Summary
+
+Add support for native TLS encryption of live migrations.
+
+## Owner
+
+*   Name: Milan Zamazal (mz-pdm)
+
+*   Email: <mzamazal@redhat.com>
+
+## Detailed Description
+
+Now, when QEMU provides native TLS encryption support, we can add support for secure migrations with a low performance overhead.  A user can enable encrypted migration for a cluster in Engine and then libvirt is asked to perform native QEMU TLS migrations when migrating VMs between hosts.
+
+## Prerequisites
+
+Cluster level 4.4 or higher is required to enable migration encryption.
+
+## Limitations
+
+Currently, an encrypted migration cannot use a separate migration network.  The destination certificate contains only the host name and not its IP addresses.  Putting IP addresses there would be cumbersome for both the code and the user (a host must be switched to the maintenance mode to update its certificate), difficult to maintain and error-prone, since the IP addresses may change at different times and places.  The migration client checks the presence of the destination address in the destination certificate, thus only the host name (and the related network) can be used as the migration destination address.
+
+A solution would be to enhance libvirt with a migration parameter that specifies the host name to check for in the certificate, independently of the destination address used.  QEMU already provides the facility, it just needs to be supported by libvirt, see this RFE: https://bugzilla.redhat.com/1754533.
+
+We will be able to remove the limitation once the libvirt RFE is implemented and released.
+
+## Benefit to oVirt
+
+Live migrations can be secured, providing more protection to the data transferred between the hosts.
+
+## User Experience
+
+There is new "Enable migration encryption" option in Migration Policy tab of the cluster edit dialog.  It's possible to enable or disable encrypted migrations there or to use the system default, which is not to encrypt migrations currently.
+
+![](/images/wiki/migration-encryption-cluster.png)
+
+There is a similar item in Host tab of the VM edit dialog, to enable or disable encrypted migrations for a particular VM if needed.
+
+![](/images/wiki/migration-encryption-vm.png)
+
+Migration encryption support is available only on cluster level 4.4 or higher.  In older clusters, it's possible to enable migration encryption for individual VMs having Custom Compatibility Version at least 4.4.
```

Comment:
not really, because the hosts won't be ready. I would remove this paragraph

Reply:
I can have a cluster < 4.4 with a couple of 4.4 hosts and migrating between them.

Reply:
yes you can, and that's exactly why it shouldn't even be mentioned, because we do not guarantee it works:)

Reply:
OK, removed. (The cluster level requirement retained, repetition doesn't hurt.:-))